src/cmap/conn/mod.rs

```diff
@@ -340,6 +377,21 @@ impl Drop for Connection {
     }
 }
 
+#[derive(Clone, Debug)]
+pub(crate) struct PinHandle {
+    connection_id: u32,
+    pool_manager: PoolManager,
+}
+
+impl PinHandle {
+    pub(crate) async fn take_connection(&self) -> Result<Connection> {
+        self.pool_manager.take_pinned(self.connection_id).await
+    }
+    pub(crate) fn unpin(&self) {
```

Comment:
can we add the `_connection` suffix here to match `take_connection`?

Reply:
Done.